swabbie-orca/src/main/kotlin/com/netflix/spinnaker/swabbie/orca/OrcaTaskMonitoringAgent.kt

```diff
@@ -0,0 +1,214 @@
+/*
+ *
+ *  * Copyright 2018 Netflix, Inc.
+ *  *
+ *  * Licensed under the Apache License, Version 2.0 (the "License")
+ *  * you may not use this file except in compliance with the License.
+ *  * You may obtain a copy of the License at
+ *  *
+ *  *   http://www.apache.org/licenses/LICENSE-2.0
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.swabbie.orca
+
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.config.SwabbieProperties
+import com.netflix.spinnaker.kork.core.RetrySupport
+import com.netflix.spinnaker.kork.eureka.RemoteStatusChangedEvent
+import com.netflix.spinnaker.kork.lock.LockManager
+import com.netflix.spinnaker.swabbie.LockingService
+import com.netflix.spinnaker.swabbie.MetricsSupport
+import com.netflix.spinnaker.swabbie.repositories.TaskCompleteEventInfo
+import com.netflix.spinnaker.swabbie.repositories.TaskTrackingRepository
+import com.netflix.spinnaker.swabbie.discovery.DiscoveryActivated
+import com.netflix.spinnaker.swabbie.events.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.stereotype.Component
+import rx.Scheduler
+import rx.schedulers.Schedulers
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+import java.time.temporal.Temporal
+import java.util.*
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicReference
+import javax.annotation.PostConstruct
+import javax.annotation.PreDestroy
+
+@Component
+class OrcaTaskMonitoringAgent (
+  private val clock: Clock,
+  val registry: Registry,
+  private val swabbieProperties: SwabbieProperties,
+  private val taskTrackingRepository: TaskTrackingRepository,
+  private val orcaService: OrcaService,
+  private val applicationEventPublisher: ApplicationEventPublisher,
+  private val retrySupport: RetrySupport,
+  private val lockingService: Optional<LockingService>
+) : DiscoveryActivated, MetricsSupport(registry) {
+
+  private val log: Logger = LoggerFactory.getLogger(javaClass)
+  private val worker: Scheduler.Worker = Schedulers.io().createWorker()
```

Comment:
Let's avoid the propagation of Rx: We're trying to get rid of it from all of our projects. Prefer the built-in Java ExecutorService.